### PR TITLE
Require VS 2022 to build this repo

### PR DESCRIPTION
**Pull request by rainersigwald**
This will be required to target .NET 6.0+ using newer SDKs and it's easier to just move.

Also disabled the Arcade default behavior to download a copy of .NET Framework MSBuild if it's not installed; instead we'll require an installed Visual Studio 17.0+.

### CODE REVIEW

**Review by rokonec — APPROVED** [review]

**Comment by Forgind** [comment]
How can someone have any VS installed but not MSBuild? Also, wouldn't this implicitly make it harder to contribute if most people are on 16.x?

**Comment by rainersigwald** [comment]
> How can someone have any VS installed but not MSBuild?

I do not understand the question, can you rephrase?

**Comment by rainersigwald** [comment]
> Also, wouldn't this implicitly make it harder to contribute if most people are on 16.x?

We target .NET 6.0 and the guidance is to use VS2022 to do that.

**Comment by Forgind** [comment]
> disabled the Arcade default behavior to download a copy of .NET Framework MSBuild if it's not installed; instead we'll require an installed Visual Studio 17.0+.

I'm wondering why this part matters. We already required VS 16.0+, right? Is there any way someone could have VS 16.0+ but not MSBuild? Or if they have MSBuild through VS, why would it matter if Arcade's default behavior doesn't do anything because they already have MSBuild?

**Comment by rainersigwald** [comment]
If VS is not installed on a Windows machine, Arcade will download a package that attempts to simulate it: https://github.com/dotnet/arcade/blob/release/6.0/Documentation/ArcadeSdk.md#globaljson

This PR opts out of that behavior.

**Comment by Nirmal4G** [comment]
Do we need to update any references from…?

1. `%ProgramFiles(x86)%\Microsoft Visual Studio\` to `%ProgramFiles%\Microsoft Visual Studio\`
2. `%SystemDrive%\Program Files (x86)\Microsoft Visual Studio\` to `%SystemDrive%\Program Files\Microsoft Visual Studio\`

**Comment by rainersigwald** [comment]
@Nirmal4G what are you thinking of, specifically?

**Comment by sharwell** [comment]
💡 You can also update this:
https://github.com/dotnet/msbuild/blob/90275e45541681fa48c1acc0fffe4bd23821ce43/MSBuild.sln#L5

**Comment by Nirmal4G** [comment]
@rainersigwald Like, if there are docs/wiki pages mentioning these paths for whatever reason, do we need to check and update there too?

@sharwell Most projects are Sdk-style so, the Min version should be `15.0` right?

**Comment by sharwell** [comment]
@Nirmal4G if the minimum version is Visual Studio 2022 (which it is from this pull request), the minimum version should be coded as `17.0.31903.59` (17.0.0).

**Comment by Nirmal4G** [comment]
@sharwell I think the Min version is for the VS solution and project system to determine if the project can be opened without triggering an upgrade wizard, like with older projects opening in newer VS versions. OR at least that's what it represents to me. Maybe it is different now!?

**Comment by sharwell** [comment]
No, it will completely block earlier versions from attempting to load the solution. You'll just get an error message saying the IDE is too old.

**Comment by rainersigwald** [comment]
> 💡 You can also update this:
> 
> https://github.com/dotnet/msbuild/blob/90275e45541681fa48c1acc0fffe4bd23821ce43/MSBuild.sln#L5

Is there any tooling that does this, or is it just a manual operation? I did it manually since I couldn't find a button for it in VS. And obviously it hadn't been touched in quite a while.

> @rainersigwald Like, if there are docs/wiki pages mentioning these paths for whatever reason, do we need to check and update there too?

I don't strongly object to that but there's a transition period since people might still be using/troubleshooting VS2019 and it's not functional, so I decided not to do it in this PR.

File: MSBuild.sln [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ﻿
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio Version 16
-VisualStudioVersion = 16.0.30413.136
-MinimumVisualStudioVersion = 10.0.40219.1
+# Visual Studio Version 17
+VisualStudioVersion = 17.0.31903.59
+MinimumVisualStudioVersion = 17.0.31903.59
```
Comment (Nirmal4G) [inline MSBuild.sln:5]: Using `save as…` within Visual Studio IDE causes it to revert to older version. May be VS bug? 🧐

Comment (sharwell) [reply MSBuild.sln:5]: > save as…

When would you Save As on a solution file?

**Comment by sharwell** [comment]
> Is there any tooling that does this, or is it just a manual operation?

Only a manual operation.
